sensor/common/metrics/metrics.go

```diff
@@ -285,6 +288,22 @@ var (
 		},
 		[]string{"central_id", "hosting", "install_method", "sensor_id"},
 	)
+
+	// responsesChannelOperationCount a counter to track the operations in the responses channel
+	responsesChannelOperationCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.SensorSubsystem.String(),
+		Name:      "num_messages_waiting_for_transmission_to_central",
+		Help:      "A counter that tracks the operations in the responses channel",
+	}, []string{"Operation"})
+
+	// responsesChannelDroppedCount keeps track of the number of messages dropped in the responses channel.
+	responsesChannelDroppedCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: metrics.PrometheusNamespace,
+		Subsystem: metrics.SensorSubsystem.String(),
+		Name:      "responses_channel_dropped_total",
+		Help:      "A counter for the total number of messages that were dropped if the responses channel is full",
+	}, []string{"MessageType"})
```

Comment:
Could this metric be integrated with `responsesChannelOperationCount` as an operation `Drop`?

The `responsesChannelOperationCount` could also benefit from adding the `MessageType` label. It would give us much more insights and should not add cost if we know the types while adding and removing from the queue.

Reply:
Good suggestion! I've merged both metrics in the last push.